kernel/sched.c

```diff
@@ -1,29 +1,34 @@
 /*
- * Copyright (c) 2016-2017 Wind River Systems, Inc.
+ * Copyright (c) 2018 Intel Corporation
  *
  * SPDX-License-Identifier: Apache-2.0
  */
-
 #include <kernel.h>
-#include <kernel_structs.h>
-#include <atomic.h>
 #include <ksched.h>
-#include <wait_q.h>
-#include <misc/util.h>
-#include <syscall_handler.h>
+#include <spinlock.h>
+#include <sched_priq.h>
+#include <timeout_q.h>
 #include <kswap.h>
+#include <kernel_arch_func.h>
+#include <syscall_handler.h>
 
 /* the only struct _kernel instance */
-struct _kernel _kernel = {0};
+struct _kernel _kernel;
 
-#ifndef CONFIG_SMP
-extern k_tid_t const _idle_thread;
-#endif
+static struct k_spinlock sched_lock;
 
-static inline int _is_thread_dummy(struct k_thread *thread)
-{
-	return _is_thread_state_set(thread, _THREAD_DUMMY);
-}
+/* Semi-experimental spin locking API using block scope.  Works great
```

Comment:
I think the final version of this patch should have more confident language than "semi-experimental".
I don't have an opinion on whether this needs to be moved to spinlock.h I'm fine with it here.